lib/openhab/dsl/rules/triggers/conditions/generic.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module OpenHAB
+  module DSL
+    module Rules
+      module Triggers
+        # @!visibility private
+        module Conditions
+          #
+          # This creates trigger conditions that work on procs
+          #
+          class Generic
+            #
+            # Create a new Condition that executes only if procs return true
+            # @param [#===, nil] from Value to check against `from` state
+            # @param [#===, nil] to Value to check against `to` state
+            # @param [#===, nil] command Value to check against received command
+            #
+            def initialize(from: nil, to: nil, command: nil)
+              @from = from
+              @to = to
+              @command = command
+            end
+
+            ANY = Generic.new.freeze # this needs to be defined _after_ initialize so its instance variables are set
+
+            #
+            # Process rule
+            # @param [Hash] inputs inputs from trigger
+            #
+            def process(mod:, inputs:) # rubocop:disable Lint/UnusedMethodArgument - mod is unused here but required
```

Comment:
It seems to me that this should be named `satisfied?` | `met?` | `match?`, but I see that Duration condition uses `process` which makes a lot of sense for the Duration's case.

Reply:
yeah, that's the interface for usage of Conditions objects from AutomationRule